src/base/backend_extended.ml

```diff
@@ -196,27 +162,19 @@ struct
     let ( / ) = div
   end
 
-  module Cvar = struct
-    include Cvar.Make (Field)
-
-    let var_indices t =
-      let _, terms = to_constant_and_terms t in
-      List.map ~f:(fun (_, v) -> v) terms
-
-    let to_constant : t -> Field.t option = function
-      | Constant x ->
-          Some x
-      | _ ->
-          None
-  end
+  module Cvar = Cvar
 
   module Constraint = struct
     open Constraint
     include Constraint.T
 
     type 'k with_constraint_args = ?label:string -> 'k
 
-    type t = (Cvar.t, Field.t) Constraint.t [@@deriving sexp]
+    type t = (Cvar.t, Field.t) Constraint.t
+
+    let t_of_sexp _ = failwith "unimplemented"
```

Comment:
I hope this is OK

Reply:
We should delete `sexp` from the interface, can be in a separate PR